transaction.go

```diff
@@ -0,0 +1,158 @@
+package seth
+
+import (
+	"bytes"
+	"math/big"
+)
+
+type Encoding int
+
+const (
+	LIST Encoding = iota
+	STRING
+	INTEGER
+)
+
+type SignedTransaction struct {
+	Tx  *Transaction
+	Sig *Signature
+}
+
+func intBytes(n Uint64) []byte {
+	var buf bytes.Buffer
+	shift := uint64(0)
+
+	for (n >> shift) > 0xFF {
+		shift += 8
+	}
+
+	for shift > 0 {
+		buf.Write([]byte{byte(n >> shift)})
+		shift -= 8
+	}
+	buf.Write([]byte{byte(n)})
+
+	return buf.Bytes()
+}
+
+func Encode(b interface{}, e Encoding) []byte {
```

Comment:
I'd un-export this function, and break it into three functions structured like
```go
func rlpWriteInt(io.Writer, Uint64)
func rlpWriteString(io.Writer, []byte)
func rlpWriteList(io.Writer, []byte)
```
which will obviate the need for defining top-level constants here, as well as the reflection.

Reply:
I agree, let's not export any general purpose RLP stuff in the top-level `seth` package. Either we should split it into a subpackage called `rlp`, or it should remain unexported.